core/lib/zksync_core/src/lib.rs

```diff
@@ -1122,6 +1123,20 @@ async fn add_house_keeper_to_task_futures(
         task_futures.push(tokio::spawn(task));
     }
 
+    if house_keeper_config.fri_gpu_prover_archiver_enabled() {
+        let fri_gpu_prover_jobs_archiver = FriGpuProverArchiver::new(
+            prover_connection_pool.clone(),
+            house_keeper_config
+                .fri_gpu_prover_archiver_reporting_interval_secs
+                .unwrap(),
+            house_keeper_config
+                .fri_gpu_prover_archiver_archiving_interval_secs
+                .unwrap(),
```

Comment:
bikeshedding: here we check the fact that if `fri_gpu_prover_archiver_enabled` is true, then both fields are `Some` in runtime. We could organize it in a single method that would encapsulate this logic and avoid unwrtaps. We could do

```
if Some((reporting_interval_secs, archiving_interval_secs)) = house_keeper_config.fri_gpu_prover_archiver_params {
//component enabled
} else {
// component disabled
}
```